File: Samples CSharp/Advanced/NavigatorFind/NavigatorFindForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Therefore.API;

namespace NavigatorFind
{
    public partial class NavigatorFindForm : Form
    {
        public NavigatorFindForm()
        {
            InitializeComponent();
        }

        private void btnFind_Click(object sender, EventArgs e)
        {
            try
            {
                TheServer server = new TheServer();
                server.Connect(TheClientType.NavigatorClient);

                TheIndexData ixData = new TheIndexData();
                ixData.SetCategory("Files", server);
                ixData["Extension"] = this.cmbExtension.SelectedItem.ToString();
                ixData["From_Folder"] = this.txtFolder.Text;

                TheApplication app = new TheApplication();
                app.NavigatorFind(ixData, TheNavigatorFindFlags.ExecuteQuery, 1);

                server.Disconnect();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        } 
    }
}

File: Samples VB.NET/Advanced/NavigatorFind/NavigatorFindForm.vb
Imports System.ComponentModel
Imports System.Text
Imports Therefore.API

Namespace NavigatorFind
	Partial Public Class NavigatorFindForm
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnFind_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnFind.Click
			Try
				Dim server As New TheServer()
				server.Connect(TheClientType.NavigatorClient)

				Dim ixData As New TheIndexData()
				ixData.SetCategory("Files", server)
				ixData("Extension") = Me.cmbExtension.SelectedItem.ToString()
				ixData("From_Folder") = Me.txtFolder.Text

				Dim app As New TheApplication()
				app.NavigatorFind(ixData, TheNavigatorFindFlags.ExecuteQuery, 1)

				server.Disconnect()
			Catch ex As Exception
				MessageBox.Show(ex.Message)
			End Try
		End Sub
	End Class
End Namespace
